src/openfermion/ops/_interaction_tensor.py

```diff
@@ -250,31 +270,21 @@ def __iter__(self):
         if self.constant:
             yield []
 
-        # 1-body elements.
-        for p in range(self.n_qubits):
-            for q in range(self.n_qubits):
-                if self.one_body_tensor[p, q]:
-                    yield [p, q]
-
-        # 2-body elements.
-        for p in range(self.n_qubits):
-            for q in range(self.n_qubits):
-                for r in range(self.n_qubits):
-                    for s in range(self.n_qubits):
-                        if self.two_body_tensor[p, q, r, s]:
-                            yield [p, q, r, s]
+        # n-body elements
+        for n, n_body_tensor in self.n_body_tensors.items():
+            for index in itertools.product(range(self.n_qubits), repeat=2 * n):
```

Comment:
👍

Reply:
Okay, I will get around to it when I have time. Thanks for checking my code!